tests/collection/collection-layout.spec.ts

```diff
@@ -1,109 +1,125 @@
+    await expect(collectionBrowser.page).toHaveURL(urlPattern);
   });
 });
 
 test(`Sort by Date published in List view`, async ({ collectionPage }) => {
+  const { collectionBrowser, infiniteScroller, sortBar } = collectionPage
+  const sortOrder = 'ascending';
+  const oppositeSortText = 'descending';
   await test.step('Switch to list view mode', async () => {
-    await collectionPage.infiniteScroller.clickViewMode(LayoutViewModeLocator.LIST);
+    await infiniteScroller.clickViewMode(LayoutViewModeLocator.LIST);
   });
 
-  await test.step('Sort by Date published - descending order', async () => {
-    await collectionPage.sortBar.applySortFilter('Date published');
-    await collectionPage.sortBar.clickSortDirection('descending');
+  await test.step(`Sort by Date published - ${sortOrder} order`, async () => {
+    await sortBar.applySortFilter('Date published');
+    await sortBar.clickSortDirection(sortOrder);
+    await expect(sortBar.srSortText).toContainText(`Change to ${oppositeSortText} sort`);
   });
 
   await test.step('Check the first 10 results if sort filters were applied', async () => {
-    await collectionPage.infiniteScroller.validateSortingResults(
-      'Date published', 'descending', 10,
-    );
-    await collectionPage.collectionBrowser.validateURLParamsWithSortFilter(
-      'Date published', 'descending',
-    );
+    const isSortedCorrectly = await infiniteScroller.validateSortingResults('Date published', sortOrder, 10);
+    expect(isSortedCorrectly).toBeTruthy();
+    const urlPattern = await collectionBrowser.urlParamsWithSortFilter('Date published', sortOrder);
+    await expect(collectionBrowser.page).toHaveURL(urlPattern);
   });
 });
 
 test(`Sort by Date archived (ascending) in Compact view`, async ({ collectionPage }) => {
+  const { collectionBrowser, sortBar } = collectionPage
+  const sortOrder = 'ascending';
   await test.step('Switch to compact view mode', async () => {
     await collectionPage.infiniteScroller.clickViewMode(LayoutViewModeLocator.COMPACT);
   });
 
-  await test.step('Sort by Date archived - ascending order', async () => {
-    await collectionPage.sortBar.applySortFilter('Date archived');
-    await collectionPage.sortBar.clickSortDirection('ascending');
+  await test.step(`Sort by Date archived - ${sortOrder} order`, async () => {
+    await sortBar.applySortFilter('Date archived');
+    await sortBar.clickSortDirection(sortOrder);
   });
 
   await test.step('Check list column headers for sort filter', async () => {
-    await collectionPage.collectionBrowser.validateCompactViewModeListLineDateHeaders(
-      'Date archived',
-    );
-    await collectionPage.collectionBrowser.validateURLParamsWithSortFilter(
-      'Date archived', 'ascending',
-    );
+    await collectionBrowser.validateCompactViewModeListLineDateHeaders('Date archived');
+    await collectionBrowser.urlParamsWithSortFilter('Date archived', sortOrder);
```

Comment:
Q: Since this method now just returns a regexp, don't we need another step to actually test correctness?